iothub/device/src/ClientFactory.cs

```diff
@@ -146,6 +136,10 @@ public static InternalClient Create(string hostname, IAuthenticationMethod authe
             }
 
             var connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
+
+            // If the file upload transport settings hasn't been specified, we will create one using the client certificate on the connection string if applicable.
+            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, ref options);
```

Comment:
here, and below: this operation is required only when the client's auth mechanism is x509 certs, it wouldn't be relevant for sas token based authentication flows.